modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java

```diff
@@ -342,6 +343,9 @@ private boolean stopping() {
             return;
 
         try {
+            if (cctx.wal() != null)
+                cctx.wal().log(new ClusterSnapshotRecord(snpName));
```

Comment:
Is WAL will be flushed after checkpoint write lock?

Reply:
Fixed.